Makefile doesn't use rebar and is broken for >R17

```
$ make
mkdir -p ebin
cp -f -- src/eredis.app.src ebin/eredis.app
sed "s/{{EXTRA_OPTS}}//" Emakefile.src > Emakefile
erl -noinput -eval 'up_to_date = make:all()' -s erlang halt
Recompile: src/eredis_sub_client
include/eredis.hrl:24: type queue() undefined
{"init terminating in do_boot",{{badmatch,error},[{erl_eval,expr,3,[]}]}}

Crash dump is being written to: erl_crash.dump...done
init terminating in do_boot ()
Makefile:8: recipe for target 'compile' failed
make: *** [compile] Error 1
$ rebar compile
==> eredis (compile)
Compiled src/basho_bench_driver_erldis.erl
Compiled src/eredis_sub_client.erl
Compiled src/eredis_client.erl
Compiled src/eredis_sub.erl
Compiled src/basho_bench_driver_eredis.erl
Compiled src/eredis.erl
Compiled src/eredis_parser.erl
```

BTW embeded `rebar` is quite outdated.
```
$ ./rebar --version
rebar version: 2 date: 20110304_023706 vcs: git d725e5f
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,6 +2,8 @@ language: erlang
 notifications:
   email: false
 otp_release:
+   - 18.0
+   - 17.5
    - 17.4
    - 17.3
    - 17.1

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,7 @@
 APP=eredis
 
+PRE17 := $(shell ERL_FLAGS="" erl -eval 'io:format("~s~n", [case re:run(erlang:system_info(otp_release), "^R") of nomatch -> ""; _ -> pre17 end]), halt().' -noshell)
+
 .PHONY: all compile clean Emakefile
 
 all: compile
@@ -20,6 +22,10 @@ endif
 
 ifdef TEST
 EXTRA_OPTS:=$(EXTRA_OPTS) {d,'TEST', true},
+endif
+
+ifndef PRE17
+EXTRA_OPTS:=$(EXTRA_OPTS) {d,namespaced_types},
 endif
 
 Emakefile: Emakefile.src

